### PR TITLE
Use a test JWT in browser tests

### PR DESCRIPTION
**Pull request by nikomakela**
KK-1168 KK-1195.

Fix the browser tests, since they have been failing after we lost an old
authentication service that was specially created for the browser test
purposes. Especially now when we have been moving (and have actually
moved already) towards the Helsinki-Profile and it's Keycloak service,
we needed another approach to authenticate our browser test user.

The OIDC client and login provider are heavily trusting in that the
profile can be read from the session storage. This set of changes tries
to mock and mimic that so that the acutaly Keycloak won't be used at all
and a symmetrically signed test JWT is used instead.

1. Mock out the Keycloak authorization by mocking data to session
storage.
2. Use a symmetrically signed test JWT for Kukkuu API requests.

Depends on https://github.com/City-of-Helsinki/kukkuu/pull/391. Also the pipeline configuration needs to provide the new browser testing configurations, e.g. the shared secret.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

File: browser-tests/utils/settings.ts [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
-dotenv.config({ path: '.env.local' });
+const browserTestEnvFilePath = '.env.test.local';
+
+if (fs.existsSync(browserTestEnvFilePath)) {
+  // eslint-disable-next-line no-console
+  console.info(
+    `Using "${browserTestEnvFilePath}" to populate environment variables for browser tests.`
+  );
+  dotenv.config({ path: browserTestEnvFilePath });
+}
```
Comment (nikomakela) [inline browser-tests/utils/settings.ts:13]: There could be an else-statement, that infos local-users, that they could use this file.

**Comment by nikomakela** [comment]
Currently in draft mode. Still to do:

- [ ]  Provide documentation to setup the testing environment (locally).
- [ ]  Configure the CI pipelines
- [ ]  Merge the API changes, so the CI pipeline for this starts working

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr577.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr577.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr577.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr577.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr577.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr577.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr577.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr577.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr577.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr577.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr577.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr577.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr577.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-ui-pr577.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**